twilight-validate/src/component.rs

```diff
@@ -674,6 +724,45 @@ pub fn select_menu(select_menu: &SelectMenu) -> Result<(), ComponentValidationEr
         self::component_select_min_values(usize::from(min_values))?;
     }
 
+    if let Some(default_values) = select_menu.default_values.as_ref() {
+        if !matches!(
+            select_menu.kind,
+            SelectMenuType::User
+                | SelectMenuType::Role
+                | SelectMenuType::Mentionable
+                | SelectMenuType::Channel
+        ) {
+            return Err(ComponentValidationError {
+                kind: ComponentValidationErrorType::SelectUnsupportedDefaultValues {
+                    kind: select_menu.kind.clone(),
+                },
+            });
+        }
+        let provided = default_values.len();
```

Comment:
sort of a nit but we dont need this variable since `.len()` is basically free and we're not modifying `default_values`

Reply:
I've moved the individual validation parts into separate functions to align with the rest of the component testing code. Now that this is moved to a parameter, I think inlining this field is no longer necessary. Please unresolve this conversation if I'm wrong, though.